From 1c8ce71c30b2acb8453b2b43c67052a6f9db2090 Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Wed, 10 Jun 2020 13:51:29 -0700
Subject: [PATCH 10/47] =?UTF-8?q?French=20clocks=20stopped=20for=209?=
 =?UTF-8?q?=E2=80=B221=E2=80=B3=20on=201911-03-11?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

(Thanks to Michael Deckers.)
* NEWS, theory.html (Accuracy of the tz database): Mention this.
* europe (Europe/Paris): Model the 1911-03-11 transition as
occurring at 00:09:21, not at 00:01.
---
 NEWS        |  7 +++++++
 europe      | 30 +++++++++++++++++++++++++++++-
 theory.html |  7 +++++++
 3 files changed, 43 insertions(+), 1 deletion(-)

diff --git a/NEWS b/NEWS
index e3ab402..5289d4e 100644
--- a/NEWS
+++ b/NEWS
@@ -16,6 +16,13 @@ Unreleased, experimental changes
     (Thanks to Géza Nyáry.)  Also, the 1890 transition to standard
     time was on 11-01, not 10-01 (thanks to Michael Deckers).
 
+    The 1911-03-11 French transition from +00:09:21 to +00 is now
+    modeled as occurring at 00:09:21, not at 00:01.  Legally, clocks
+    stopped at 00:00 for 9 minutes, 21 seconds but this cannot be
+    represented in tzdb, so tzdb instead represents the common
+    practice of keeping an old clock running until the new clock
+    started up.  (Thanks to Michael Deckers.)
+
   Changes to code
 
     The undocumented and ineffective tzsetwall function has been
diff --git a/europe b/europe
index f0acb42..4fa7119 100644
--- a/europe
+++ b/europe
@@ -1326,6 +1326,34 @@ Link	Europe/Helsinki	Europe/Mariehamn
 # Françoise Gauquelin, Problèmes de l'heure résolus en astrologie,
 # Guy Trédaniel, Paris 1987
 
+# From Michael Deckers (2020-06-10):
+# Le Gaulois, 1911-03-11, page 1/6, online at
+# https://www.retronews.fr/societe/echo-de-presse/2018/01/29/1911-change-lheure-de-paris
+# ... [ Instantly, all pressure driven clock dials halted...  Nine minutes and
+#       twenty-one seconds later the hands resumed their circular motion. ]
+# There are also precise reports about how the change was prepared in train
+# stations: all the publicly visible clocks stopped at midnight railway time
+# (or were covered), only the chief of service had a watch, labeled
+# "Heure ancienne", that he kept running until it reached 00:04:21, when
+# he announced "Heure nouvelle".  See the "Le Petit Journal 1911-03-11".
+# https://gallica.bnf.fr/ark:/12148/bpt6k6192911/f1.item.zoom
+#
+# From Paul Eggert (2020-06-10):
+# French time in railway stations was legally five minutes behind civil time,
+# which explains why "old time" ran to 00:04:21 instead of to 00:09:21.
+# The Brooklyn Daily Eagle Almanac (1912), page 494, says:
+#
+#   ALL CLOCKS STOPPED IN FRANCE.
+#   On March 10, 1911, all clocks in the Republic of France were stopped
+#   for 9 minutes and 21 seconds.  This was in obedience to a measure
+#   adopted by the French Senate, which went into effect at midnight....
+#   Owing to this change in time a question arose in the French press as
+#   to whether or not a child that was born and died within the elapsed
+#   time could be said to have legally lived.
+#
+# tzdb has no way to represent stopped clocks.  As the railway practice
+# was to keep a watch running on "old time" to decide when to restart
+# the other clocks, model this as a transition for "old time" at 00:09:21.
 
 #
 # Shank & Pottenger seem to use '24:00' ambiguously; resolve it with Whitman.
@@ -1395,7 +1423,7 @@ Rule	France	1976	only	-	Sep	26	 1:00	0	-
 # on PMT-0:09:21 until 1978-08-09, when the time base finally switched to UTC.
 # Zone	NAME		STDOFF	RULES	FORMAT	[UNTIL]
 Zone	Europe/Paris	0:09:21 -	LMT	1891 Mar 15  0:01
-			0:09:21	-	PMT	1911 Mar 11  0:01 # Paris MT
+			0:09:21	-	PMT	1911 Mar 11  0:09:21 # Paris MT
 # Shanks & Pottenger give 1940 Jun 14 0:00; go with Excoffier and Le Corre.
 			0:00	France	WE%sT	1940 Jun 14 23:00
 # Le Corre says Paris stuck with occupied-France time after the liberation;
diff --git a/theory.html b/theory.html
index ffa3b4d..de105f2 100644
--- a/theory.html
+++ b/theory.html
@@ -691,6 +691,13 @@ href="https://www.dissentmagazine.org/blog/booked-a-global-history-of-time-vanes
     <code><abbr>tz</abbr></code> database requires.
   </li>
   <li>
+    The <code><abbr>tz</abbr></code> database cannot represent stopped clocks.
+    However, on 1911-03-11 at 00:00, French clocks were changed by
+    stopping them for 9 minutes, 21 seconds. This is approximated
+    in <code><abbr>tz</abbr></code> as a transition from 00:09:21 back
+    to 00:00:00 that day.
+  </li>
+  <li>
     Sometimes historical timekeeping was specified more precisely
     than what the <code><abbr>tz</abbr></code> code can handle.
     For example, from 1909 to 1937 <a
-- 
1.8.3.1

